addons/netfox/rollback/rollback-synchronizer.gd

```diff
@@ -150,21 +150,35 @@ func _process_tick(tick: int):
 			NetworkRollback.process_rollback(node, NetworkTime.ticktime, tick, is_fresh)
 			_freshness_store.notify_processed(node, tick)
 
+
+
 func _record_tick(tick: int):
 	# Broadcast state we own
 	if not _auth_state_props.is_empty():
-		var broadcast = {}
-
+		var state_to_broadcast = {}
+		
+		#if (multiplayer.get_unique_id() > 1):
+			#print("Record tick state sending, peer id is: %s" % multiplayer.get_unique_id())
 		for property in _auth_state_props:
 			if _can_simulate(property.node, tick - 1):
 				# Only broadcast if we've simulated the node
-				broadcast[property.to_string()] = property.get_value()
+				state_to_broadcast[property.to_string()] = property.get_value()
 	
-		if broadcast.size() > 0:
-			# Broadcast as new state
+		if state_to_broadcast.size() > 0:
 			_latest_state = max(_latest_state, tick)
-			_states[tick] = PropertySnapshot.merge(_states.get(tick, {}), broadcast)
-			_submit_state.rpc(broadcast, tick)
+			_states[tick] = PropertySnapshot.merge(_states.get(tick, {}), state_to_broadcast)
+			
+			if (NetworkRollback.enable_state_serialization):
+				var serialized_current_state: PackedByteArray = PropertiesSerializer.serialize_multiple_properties(_auth_state_props, tick)
+				_serialized_states[tick] = serialized_current_state
+				
+				# Broadcast as new state
+				for picked_peer_id in multiplayer.get_peers():
+					_submit_serialized_state.rpc_id(picked_peer_id, serialized_current_state)
+			else:
+				# Broadcast as new state
+				for picked_peer_id in multiplayer.get_peers():
+					_submit_state.rpc_id(picked_peer_id, state_to_broadcast, tick)
```

Comment:
Same thing as inputs. Locally, not a single RPC should be sent/received. Because we already have the state locally. So we must simply RPC to other peers, because for this tick we are RPCing, we already have set locally the variables `latest_state`, `_states[tick]`

Reply:
@rpc's `call_remote` makes the old rpc call work no problems, so this is needless.